src/main/java/org/broadinstitute/hellbender/dev/pipelines/bqsr/FewReadsWriter.java

```diff
@@ -0,0 +1,78 @@
+package org.broadinstitute.hellbender.dev.pipelines.bqsr;
+
+
+import com.google.api.services.genomics.model.Read;
+import com.google.cloud.dataflow.sdk.Pipeline;
+import com.google.cloud.dataflow.sdk.transforms.Create;
+import com.google.cloud.dataflow.sdk.transforms.DoFn;
+import com.google.cloud.dataflow.sdk.transforms.ParDo;
+import com.google.cloud.dataflow.sdk.transforms.View;
+import com.google.cloud.dataflow.sdk.values.PCollection;
+import com.google.cloud.dataflow.sdk.values.PCollectionView;
+import com.google.cloud.genomics.gatk.common.GenomicsConverter;
+import htsjdk.samtools.SAMFileHeader;
+import htsjdk.samtools.SAMFileWriter;
+import htsjdk.samtools.SAMFileWriterFactory;
+import htsjdk.samtools.SAMRecord;
+import org.broadinstitute.hellbender.utils.dataflow.BucketUtils;
+
+import java.io.OutputStream;
+import java.io.Serializable;
+
+/**
+ * Takes a few Reads and will write them to a BAM file.
+ * However all the reads must fit into a single worker's memory, so this won't go well if you have too many.
+ */
+public class FewReadsWriter implements Serializable {
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * Takes a few Reads and will write them to a BAM file.
+     * However all the reads must fit into a single worker's memory, so this won't go well if you have too many.
+     */
+    public static void WriteToFile(Pipeline pipeline, PCollection<Read> reads, final SAMFileHeader header, final String filename) {
+        new FewReadsWriter().inner_WriteToFile(pipeline, reads, header, filename);
+    }
+
+    private void inner_WriteToFile(Pipeline pipeline, PCollection<Read> reads, final SAMFileHeader header, final String filename) {
+
+        PCollectionView<Iterable<Read>> iterableView =
+                reads.apply(View.<Read>asIterable());
+
+        PCollection<String> dummy = pipeline.apply(Create.<String>of(filename));
+
+        dummy.apply(ParDo.named("save to file")
+                        .withSideInputs(iterableView)
+                        .of(new SaveToFile(header, iterableView))
+        );
+
+    }
+
+    private class SaveToFile extends DoFn<String,Void> implements Serializable {
+        private static final long serialVersionUID = 1L;
+        private final SAMFileHeader header;
+        private final PCollectionView<Iterable<Read>> iterableView;
+
+        public SaveToFile(SAMFileHeader header, PCollectionView<Iterable<Read>> iterableView) {
+            this.header = header;
+            this.iterableView = iterableView;
+        }
+
+        @Override
+        public void processElement(ProcessContext c) throws Exception {
+            String dest = c.element();
+            System.out.println("Saving to "+dest);
+            Iterable<Read> reads = c.sideInput(iterableView);
+            OutputStream outputStream = BucketUtils.createFile(dest, c.getPipelineOptions());
+            SAMFileWriter writer = new SAMFileWriterFactory().makeBAMWriter(header, false, outputStream);
```

Comment:
Use try-with-resources to manage the closing of the writer (+ underlying stream) automatically